packages/aws-cdk-lib/aws-efs/lib/efs-file-system.ts

```diff
@@ -548,6 +558,10 @@ export class FileSystem extends FileSystemBase {
 
     const oneZoneAzName = props.vpc.availabilityZones[0];
 
+    const fileSystemProtection = props.replicationOverwriteProtection !== undefined ? {
+      replicationOverwriteProtection: props.replicationOverwriteProtection ? 'ENABLED' : 'DISABLED',
+    } : undefined;
```

Comment:
with enums theres no need for this check and it can be a simple passthrough 👍

Reply:
Unfortunately, it seems necessary to continue checks because an object or undefined must be passed here. 
However, the ternary operator that replaces with the strings ENABLED, DISABLED seems to be removable!